api/src/pcapi/repository/clean_database.py

```diff
@@ -23,6 +23,7 @@ def clean_all_database(*args, **kwargs):  # type: ignore [no-untyped-def]
     """Order of deletions matters because of foreign key constraints"""
     if settings.ENV not in ("development", "testing"):
         raise ValueError(f"You cannot do this on this environment: '{settings.ENV}'")
+    fraud_models.ProductWhitelist.query.delete()
```

Comment:
Je mettrais bien ça avec les autres suppression des modèles de `fraud_models`. Pour avoir tous les modèles de ce module au même endroit.

D'ailleurs, au passage, je pense qu'il manque la suppression d'autres modèles : `OrphanDmsApplication`, `BlacklistedDomainName`. Est-ce que tu pourrais amender ton commit pour supprimer aussi ces modèles (et modifier le message de commit, du genre : "Empty fraud-related tables in clean_all_database" ?

Reply:
> Je mettrais bien ça avec les autres suppression des modèles de fraud_models. Pour avoir tous les modèles de ce module au même endroit.

Je ne sais pas si c'est une bonne idée de tout vouloir rassembler par class, car la BDD est le fruit de migration dans un ordre défini qui ne s'explique pas par "modèle". C'est pourquoi j'ai choisi de rajouter tout en haut, ce qui parraissait logique (on supprime les dernières tables en premier =~ risquer moins de problème.)

> D'ailleurs, au passage, je pense qu'il manque la suppression d'autres modèles : OrphanDmsApplication, BlacklistedDomainName. Est-ce que tu pourrais amender ton commit pour supprimer aussi ces modèles (et modifier le message de commit, du genre : "Empty fraud-related tables in clean_all_database" ?

Oui pourquoi pas, je peux regarder dans un autre commit, mais est-ce qu'on doit tout ranger par module ou pas (même si je ne pense pas que ça soit une bonne idée..., sauf erreur de ma part)

Reply:
À part `BeneficiaryFraudCheck` et `BeneficiaryFraudReview` qui dépendent de `User` (et doivent donc être supprimées avant), les tables de `core.fraud.models` n'ont pas de relation avec les autres tables. Donc on peut les supprimer quand on veut. Donc je les aurais regroupées pour faciliter la lecture (et aussi la détection de modèles oubliés ! ;) ), mais ça n'est pas très important.

Reply:
`product_whitelist` a également une relation avec user via `authorId` (qui est l'opérateur a avoir autorisé le livre), je prends en compte tes remarques et te reping pour un fix.